src/org/thoughtcrime/securesms/WelcomeActivity.java

```diff
@@ -138,12 +143,37 @@ private void startImportBackup() {
                     } else {
                         final String backupFile = dcContext.imexHasBackup(imexDir.getAbsolutePath());
                         if (backupFile != null) {
+
+
+                            View gl = View.inflate(this, R.layout.dialog_with_checkbox, null);
+                            CheckBox encryptCheckbox = gl.findViewById(R.id.dialog_checkbox);
+                            TextView msg = gl.findViewById(R.id.dialog_message);
+
+                            // If we'd use both `setMessage()` and `setView()` on the same AlertDialog, on small screens the
+                            // "OK" and "Cancel" buttons would not be show. So, put the message into our custom view:
+                            msg.setText(String.format(getResources().getString(R.string.import_backup_ask), backupFile) );
+                            encryptCheckbox.setText("Encrypt database (highly experimental, use at your own risk)");
+                            int[]      tintAttr   = new int[]{android.R.attr.textColorSecondary};
+                            TypedArray typedArray = obtainStyledAttributes(tintAttr);
+                            int        color      = typedArray.getColor(0, Color.GRAY);
+                            typedArray.recycle();
+                            encryptCheckbox.setTextColor(color);
+
                             new AlertDialog.Builder(this)
                                     .setTitle(R.string.import_backup_title)
-                                    .setMessage(String.format(getResources().getString(R.string.import_backup_ask), backupFile))
+                                    .setView(gl)
                                     .setNegativeButton(android.R.string.cancel, null)
-                                    .setPositiveButton(android.R.string.ok, (dialog, which) -> startImport(backupFile, null))
+                                    .setPositiveButton(android.R.string.ok, (dialog, which) -> startImport(backupFile, null, encryptCheckbox.isChecked()))
                                     .show();
+
+
+
+//                            new AlertDialog.Builder(this)
```

Comment:
is this still needed / planned to be reused?

Reply:
No, I forgot to remove it